main.go

```diff
@@ -19,7 +19,7 @@ func main() {
 	opts := &plugin.ServeOpts{ProviderFunc: provider.New}
 
 	if debugMode {
-		err := plugin.Debug(context.Background(), "registry.terraform.io/providers/kentik/kentik-cloudexport", opts)
+		err := plugin.Debug(context.Background(), "kentik/kentik-cloudexport", opts)
```

Comment:
Have you tested that?

Reply:
Yes, I did. It seems that I have installed the provider under `kentik/kentik-cloudexport`. Looking at the Makefile, your change is correct. I wasn't sure whether `source` in our `provider.tf` files is a local one. I just need to better understand those source resolution rules.

```
terraform {
  required_providers {
    kentik-cloudexport = {
      version = ">= 0.3.0"
      source = "kentik/kentik-cloudexport" # use provider from registry.terraform.io
      # source  = "kentik/automation/kentik-cloudexport" # use locally-built provider (make install)
    }
  }
}
```
Now the question is, whether we want the debugger, by default, to connect with a local provider or the one from the registry.

Reply:
I think it's better to debug locally-build provider by default.

It looks inconsistent that we have `automation` namespace only for local provider and not for the published one. Could you analyze if it is something we could improve?

Reply:
IMHO we should remove the `automation` namespace. My argument is that I do not know why it is there :D
`kentik/kentik-cloudexport` makes sense, as it's compliant with terraform registry URLs, and I see this convention `<company>/<provider_name>` is widely adopted. Do you guys have any arguments for keeping the `automation` namespace, for locally-built providers? @Fenthick @mmac-m3a @mateuszmidor

Reply:
We have `automation` namespace, because setting it was showed in example provider (https://github.com/hashicorp/terraform-provider-hashicups/blob/596cefab4a4d23c664f3b17494db431b599becce/Makefile#L3).

If we don't need this namespace and kentik-cloudexport is published without it, I don't see a reason to keep it in local installation.

Reply:
Just now while happily removing the `automation` namespace I understood that we need a way to distinguish between a published and local provider, and the sole purpose of that namespace is to enable us to do that.

> Could you analyze if it is something we could improve?

Analysis complete. It is already perfect.

Reply:
I just found that the namespace of published provider is `kentik`: https://registry.terraform.io/namespaces/kentik

Looked at other providers. Notice that Akamai uses `akamai` as namespace both in registry and local build: https://github.com/akamai/terraform-provider-akamai/blob/master/GNUmakefile#L8 They also use `registry.terraform.io` as a registry hostname.

> I understood that we need a way to distinguish between a published and local provider, and the sole purpose of that namespace is to enable us to do that.

Yes, this is convenient. There are other ways to do that (https://www.terraform.io/docs/cli/config/config-file.html#development-overrides-for-provider-developers), but having different source does the job.

I wouldn't call current situation perfect, but good enough.